### PR TITLE
feat(order status): draft version of model

### LINKED ISSUE #113: [New Model]: OrderStatus
### Context
PURIS provides data as an operationalization of a DCM planning. You want to verify that the material needed for your production will be available on time. Therefore you commonly match the customer's demand against the supplier's production outputs. Also the customer's material stock (used to manufacture the products) matches the supplier's product stock (produced goods of the supplier).

### Actual Model
As these information are exchanged for an existing business relationship between two partner, we use the order (from Supplier and Customer side of view) as an "anchor point". The OrderStatus represents the **LATEST** status of the Order from **SUPPLIER** point of view. It has multiple positions that contains "AllocatedStocks". The AllocatedStock represents a quantity of a material, that has already been produced but not yet shipped.

## Relevant Standards
None

## Example Data
{
  "manufacturerOrderId": "M-Nbr-4711",
  "customerOrderId": "C-Nbr-4711",
  "orderPositionStatuses": [
    {
      "requiredDateOfCompletion": "2023-03",
      "manufacturerPartId": "123-0.740-3434-A",
      "customerSiteBpns": "BPNS1234567890YY",
      "allocatedStocks": [
        {
          "quantityOnAllocatedStock": {
            "quantityNumber": 2.5,
            "measurementUnit": "litre"
          },
          "manufacturerStockSiteBpns": "BPNS1234567890YY"
        }
      ],
      "positionId": "PositionId-01",
      "quantityOrdered": {
        "quantityNumber": 2.5,
        "measurementUnit": "litre"
      },
      "totalAllocatedQuantityOnStock": {
        "quantityNumber": 2.5,
        "measurementUnit": "litre"
      }
    }
  ]
}

## MS1 Criteria

- [ ] The proposed aspect model does not exist already
- [ ] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [ ] The proposal references a Catena-X use case
- [ ] Relevant standards are mentioned/linked
- [ ] A first draft of the data structure is provided in an example

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description
Draft version for the order status of a supplier side including stocks of material that has already been produced for the order.

Closes #113 

## MS2 Criteria
(to be filled out by PR reviewer)
- [ ] the model **validates** with the BAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar bamm-cli.jar -i \<path-to-aspect-model\> -v ). The  BAMM CLI is available [here](https://openmanufacturingplatform.github.io/sds-documentation/sds-developer-guide/dev-snapshot/tooling-guide/bamm-cli.html) and in [GitHub](https://github.com/OpenManufacturingPlatform/sds-sdk/releases)
- [ ] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [ ] the identifiers for all model elements **start with a capital letter** except for properties
- [ ] the identifier for **properties starts with a small letter**
- [ ] all model elements **at least contain the fields "name" and "description"** in English language. 
- [ ] **no duplicate names or preferredNames** within an Aspect (e.g. a Property and the referenced Characteristic should not have the same name)
- [ ] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [ ] use **abbreviations only when necessary** and if these are sufficiently common
- [ ] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [ ] fields `preferredName` and `description` are not the same
- [ ] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [ ] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [ ] units are referenced from the BAMM unit catalog whenever possible
- [ ] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [ ] when relying on **external standards**, they are referenced through a **"see"** element
- [ ] all properties with an [simple type](https://openmanufacturingplatform.github.io/sds-documentation/bamm-specification/v1.0.0/datatypes.html) have an example value
- [ ] metadata.json exists with status "release"
- [ ] file RELEASE_NOTES.md exists and contains entries for proposed model changes 

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [ ] All required reviewers have approved this PR (see reviewers section)
- [ ] The new aspect (version) will be implemented by at least one data provider
- [ ] The new aspect (version) will be consumed by at least one data consumer
- [ ] There exists valid test data
- [ ] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [ ] The model has at least version '1.0.0'

### CODE REVIEW

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+:OrderStatus a bamm:Aspect;
+    bamm:properties (:manufacturerOrderId [
+  bamm:property :customerOrderId;
+  bamm:optional "true"^^xsd:boolean
+] :orderPositionStatuses);
+    bamm:operations ();
+    bamm:events ();
+    bamm:preferredName "Status of Order"@en;
+    bamm:description "This aspect represents the latest status of an order, that describes the business relationship between a customer and a manufacturer along the value chain. "@en.
+:manufacturerOrderId a bamm:Property;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:22]: There is an ongoing discussion on defining shared aspects for several catenax specific ids. Maybe manufacturerOrderId is one and could be used here.

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    bamm:description "This is the order number on the manufacturer side that is given by e.g., the manufacturer's enterprise resource planning system."@en;
+    bamm:characteristic :OrderIdCharacteristic;
+    bamm:exampleValue "M-Nbr-4711".
+:customerOrderId a bamm:Property;
+    bamm:preferredName "Customer Order ID"@en;
+    bamm:description "This is the order number on the customer side that can be given by e.g., the customer's enterprise resource planning system."@en;
+    bamm:characteristic :OrderIdCharacteristic;
+    bamm:exampleValue "C-Nbr-4711".
+:orderPositionStatuses a bamm:Property;
+    bamm:preferredName "Order Position Statuses"@en;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:33]: same here. Not sure, if statuses is the correct plural

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    bamm:description "One Order consists of one or more Order Positions. Therefore the OrderStatus consists of one OrderPositionStatus per OrderPosition (one to one matching).\n"@en;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:34]: Upper and lower case writing (Order -> order)

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+:OrderPositionStatusSortedSet a bamm-c:SortedSet;
+    bamm:description "Each OrderPositionStatus must be unique. The OrderPositionStatus within the Set are sorted by the position id."@en;
+    bamm:dataType :OrderPositionStatus.
+:OrderPositionStatus a bamm:Entity;
+    bamm:properties (:positionId :totalAllocatedQuantityOnStock :quantityOrdered :customerSiteBpns [
+  bamm:property :allocatedStocks;
+  bamm:optional "true"^^xsd:boolean
+] :requiredDateOfCompletion ext-planned:manufacturerPartId);
+    bamm:preferredName "Status of Order Position"@en;
+    bamm:description "An Order Position is described by a quantity of one material (PartAsPlanned), which is requested by customer's site to be manufactured until a given time.\nThe Order Position Status includes the production status of the Order Position including already manufactured material for the Order Position."@en.
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:48]: ... is requested by a customer to be manufactured ...

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+:positionId a bamm:Property;
+    bamm:preferredName "Position ID"@en;
+    bamm:description "This is the business identifier for an Order Position."@en;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:51]: is business identifier a given concept? I haven't heard about that

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    bamm:preferredName "Total Allocated Quantity on Stock"@en;
+    bamm:description "The total Stock is the sum of all Material Quanities in AllocatedStocks related to this position (OrderPositionStatus.allocatedStocks)."@en;
+    bamm:characteristic ext-relationship:QuantityCharacteristic.
+:quantityOrdered a bamm:Property;
+    bamm:preferredName "Total Quantity Ordered"@en;
+    bamm:description "This is the quantity ordered by the customer for this position."@en;
+    bamm:characteristic ext-relationship:QuantityCharacteristic.
+:customerSiteBpns a bamm:Property;
+    bamm:preferredName "Bussiness Partner Number Site of Customer Site"@en;
+    bamm:description "To check the naming in Call-off standard"@en;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:64]: Don't get the description

File: io.catenax.order_status/1.0.0/OrderStatus.ttl [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    bamm:characteristic :BusinessPartnerNumberSiteCharacteristic;
```
Comment (bs-jokri) [inline io.catenax.order_status/1.0.0/OrderStatus.ttl:65]: Is this the BPN? If so, there will also be a separate shared aspect, which might be used here

File: io.catenax.order_status/README.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (bs-jokri) [inline io.catenax.order_status/README.md:1]: rename file to "RELEASE_NOTES.md" as with the other models

File: io.catenax.order_status/1.0.0/OrderStatus.ttl (hunk 7 above)
Comment (tom-rm-meyer-ISST) [reply io.catenax.order_status/1.0.0/OrderStatus.ttl:65]: Ëxactly, it'll be a BPN - or more concretely a BPNS.

File: io.catenax.order_status/1.0.0/metadata.json [hunk 9]
```diff
@@ -0,0 +1 @@
+{ status: draft }
```
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply io.catenax.order_status/1.0.0/metadata.json:1]: Shall I already update it, or is this just meant as a reminder? BTW, I updated it anyway as I missed to set the quotation marks in JSON.

File: io.catenax.order_status/1.0.0/OrderStatus.ttl (hunk 6 above)
Comment (tom-rm-meyer-ISST) [reply io.catenax.order_status/1.0.0/OrderStatus.ttl:64]: Ops, that was an open todo. "Site of the customer business partner, that will receive the materials."

File: io.catenax.order_status/1.0.0/OrderStatus.ttl (hunk 5 above)
Comment (tom-rm-meyer-ISST) [reply io.catenax.order_status/1.0.0/OrderStatus.ttl:51]: The term "business identifier" shall indicate that this id is not a technical one (UUID) but one that is used to refer to the position. From that understanding the manufacturerPartId would also be a business identifier.
Let's quickly discuss later on.

Comment (tom-rm-meyer-ISST) [reply io.catenax.order_status/1.0.0/OrderStatus.ttl]: I updated the description as aligned during discussion.

**Comment by tom-rm-meyer-ISST** [comment]
Replaced via new model [issue#145](https://github.com/eclipse-tractusx/sldt-semantic-models/issues/145)

**Pull request closed without merging**